src/edu/csus/ecs/pc2/core/report/ExportFilesUtiltiites.java

```diff
@@ -0,0 +1,88 @@
+package edu.csus.ecs.pc2.core.report;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+
+import edu.csus.ecs.pc2.core.Constants;
+import edu.csus.ecs.pc2.core.FileUtilities;
+import edu.csus.ecs.pc2.core.execute.ExecuteUtilities;
+import edu.csus.ecs.pc2.core.imports.clics.CLICSAward;
+import edu.csus.ecs.pc2.core.imports.clics.CLICSAwardUtilities;
+import edu.csus.ecs.pc2.core.log.StaticLog;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.exports.ccs.ResultsFile;
+import edu.csus.ecs.pc2.services.core.ScoreboardJson;
+
+public class ExportFilesUtiltiites {
+
+    /**
+     * Write contest results files to directory.
+     * 
+     * Creates contest result files
+     * <li>{@link ResultsFile.RESULTS_FILENAME}
+     * <li>{@link Constants.SCOREBOARD_JSON_FILENAME}
+     * <li>{@link Constants.AWARDS_JSON_FILENAME}
+     * 
+     * @param contest
+     * @param outputDirectory
+     *            directory to write reports to.
+     * @return list of created files
+     */
+    public static String[] writeResultsFiles(IInternalContest contest, String outputDirectory) {
+
+        List<String> filesCreated = new ArrayList<String>();
+
+        String resultsFilename = outputDirectory + File.separator + ResultsFile.RESULTS_FILENAME;
+
+        String scoreboardJsonFilename = outputDirectory + File.separator + Constants.SCOREBOARD_JSON_FILENAME;
+
+        String awardsFileName = outputDirectory + File.separator + Constants.AWARDS_JSON_FILENAME;
+
+        try {
+            ResultsFile resultsFile = new ResultsFile();
+            String[] resultTSVLines = resultsFile.createTSVFileLines(contest);
+            FileUtilities.writeFileContents(resultsFilename, resultTSVLines);
+
+            filesCreated.add(resultsFilename);
+
+        } catch (Exception e) {
+            StaticLog.getLog().log(Level.WARNING, "Problem writing results file", e);
+            e.printStackTrace(); // TODO 760 how to report problem to user ?
+            throw ExecuteUtilities.rethrow(e);
```

Comment:
I don't understand the above statement - specifically, why it contains the word `throw`.  The call to `ExecuteUtilities.rethrow(e)` will result in that called method throwing an exception -- which seems to me means that it will never return here to execute the `throw` in the above statement (so, why is it there?)

Maybe I just don't understand Java Exception Handling as well as I thought I did...

Reply:
re: rethrow.   I've seen that used before in other systems.
The intent of rethrow is to throw an exception without requiring a throws clause in the method where it is used.  
If there is a more clear name, let us know.